packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -128,41 +152,91 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
       };
 
       await program.host.writeFile(
-        resolvePath(generatedFolder, configurationFileName),
+        resolvePath(outputFolder, configurationFileName),
         prettierOutput(JSON.stringify(configurations, null, 2))
       );
 
       if (options.skipSDKGeneration !== true) {
-        const csProjFile = resolvePath(outputFolder, `${configurations["library-name"]}.csproj`);
+        const csProjFile = resolvePath(
+          outputFolder,
+          "src",
+          `${configurations["library-name"]}.csproj`
+        );
         logger.info(`Checking if ${csProjFile} exists`);
+        const newProjectOption = "";
+        // TODO uncomment when https://github.com/Azure/autorest.csharp/issues/4463 is resolved
+        //  options["new-project"] || !existsSync(csProjFile) ? "--new-project" : "";
+        const existingProjectOption = options["existing-project-folder"]
+          ? `--existing-project-folder ${options["existing-project-folder"]}`
+          : "";
+        const debugFlag = options.debug ?? false ? " --debug" : "";
 
-        logger.info("TODO connect the dotnet generator");
-        //const command = `dotnet --roll-forward Major ${resolvePath(
-        //  options.csharpGeneratorPath
-        //)} --project-path ${outputFolder} ${newProjectOption} ${existingProjectOption} --clear-output-folder ${
-        //  options["clear-output-folder"]
-        //}${debugFlag}`;
-        //logger.info(command);
-        //
-        //try {
-        //  execSync(command, { stdio: "inherit" });
-        //} catch (error: any) {
-        //  if (error.message) logger.info(error.message);
-        //  if (error.stderr) logger.error(error.stderr);
-        //  if (error.stdout) logger.verbose(error.stdout);
-        //  throw error;
-        //}
-      }
+        const projectRoot = findProjectRoot(dirname(fileURLToPath(import.meta.url)));
+        const generatorPath = resolvePath(
+          projectRoot + "/dist/generator/Microsoft.Generator.CSharp.exe"
```

Comment:
is `Microsoft.Generator.CSharp.exe` the name of the executable file in linux too? If we already test this in linux ci/cd then feel free to ignore.

Reply:
yeah the ci runs on windows and linux so that will be tested.

Reply:
good call out linux is failing :-).

Reply:
swapped to dll, its fixed now